storage/spanner.go

```diff
@@ -0,0 +1,984 @@
+	_, err = s.getSpannerClient(table).ReadWriteTransaction(ctx, func(ctx context.Context, t *spanner.ReadWriteTransaction) error {
+		tmpMap := map[string]interface{}{}
+		for k, v := range m1 {
+			tmpMap[k] = v
+		}
+		if len(eval.Attributes) > 0 || expr != nil {
+			status, _ := evaluateConditionalExpression(ctx, t, table, tmpMap, eval, expr)
+			if !status {
+				return errors.New("ConditionalCheckFailedException")
+			}
+		}
+		table = changeTableNameForSP(table)
+
+		r, err := t.ReadRow(ctx, table, key, cols)
+		if err != nil {
+			return errors.New("ResourceNotFoundException", err)
+		}
+		rs, err := parseRowForNull(r, colDLL, cols)
+		if err != nil {
+			return err
+		}
+		for k, v := range tmpMap {
+			v1, ok := rs[k]
+			if ok {
+				switch v1.(type) {
+				case int64:
+					v2, ok := v.(float64)
+					if !ok {
+						strV, ok := v.(string)
+						if !ok {
+							return errors.New("ValidationException", reflect.TypeOf(v).String())
+						}
+						v2, err = strconv.ParseFloat(strV, 64)
+						if err != nil {
+							return errors.New("ValidationException", reflect.TypeOf(v).String())
+						}
+						err = checkInifinty(v2, strV)
+						if err != nil {
+							return err
+						}
+					}
+					tmpMap[k] = v1.(int64) + int64(v2)
+					err = checkInifinty(float64(m[k].(int64)), m)
+					if err != nil {
+						return err
+					}
+				case float64:
+					v2, ok := v.(float64)
+					if !ok {
+						strV, ok := v.(string)
+						if !ok {
+							return errors.New("ValidationException", reflect.TypeOf(v).String())
+						}
+						v2, err = strconv.ParseFloat(strV, 64)
+						if err != nil {
+							return errors.New("ValidationException", reflect.TypeOf(v).String())
+						}
+						err = checkInifinty(v2, strV)
+						if err != nil {
+							return err
+						}
+					}
+					tmpMap[k] = v1.(float64) + v2
+					err = checkInifinty(m[k].(float64), m)
+					if err != nil {
+						return err
+					}
+
+				case []interface{}:
+					var ifaces1 []interface{}
+					ba, ok := v.([]byte)
+					if ok {
+						json.Unmarshal(ba, &ifaces1)
+					} else {
+						ifaces1 = v.([]interface{})
+					}
+					m1 := map[interface{}]struct{}{}
+					ifaces := v1.([]interface{})
+					for i := 0; i < len(ifaces); i++ {
+						m1[ifaces[i]] = struct{}{}
+					}
+					for i := 0; i < len(ifaces1); i++ {
+						m1[ifaces1[i]] = struct{}{}
+					}
+					ifaces = []interface{}{}
+					for k := range m1 {
+						ifaces = append(ifaces, k)
+					}
+					tmpMap[k] = ifaces
+				default:
+					logger.LogDebug(reflect.TypeOf(v).String())
+				}
+			}
+		}
+		tmpMap[pKey] = pValue
+		if sValue != nil {
+			tmpMap[sKey] = sValue
+		}
+		ddl := models.TableDDL[table]
+
+		for k, v := range tmpMap {
+			updatedObj[k] = v
+			t, ok := ddl[k]
+			if t == "BYTES(MAX)" && ok {
+				ba, err := json.Marshal(v)
+				if err != nil {
+					return errors.New("ValidationException", err)
+				}
+				tmpMap[k] = ba
+			}
+		}
+
+		mutation := spanner.InsertOrUpdateMap(table, tmpMap)
```

Comment:
See above. This could be an `Update` mutation.

Reply:
@rosspatil your thoughts